[2023-11-22T23:28:14.597+0100] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: xml_parsing.parse_xml scheduled__2023-11-22T22:23:00+00:00 [queued]>
[2023-11-22T23:28:14.603+0100] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: xml_parsing.parse_xml scheduled__2023-11-22T22:23:00+00:00 [queued]>
[2023-11-22T23:28:14.604+0100] {taskinstance.py:1361} INFO - Starting attempt 4 of 6
[2023-11-22T23:28:14.622+0100] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): parse_xml> on 2023-11-22 22:23:00+00:00
[2023-11-22T23:28:14.625+0100] {standard_task_runner.py:57} INFO - Started process 24037 to run task
[2023-11-22T23:28:14.628+0100] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'xml_parsing', 'parse_xml', 'scheduled__2023-11-22T22:23:00+00:00', '--job-id', '80', '--raw', '--subdir', 'DAGS_FOLDER/xml_parsing_DAG.py', '--cfg-path', '/tmp/tmpm_owecgn']
[2023-11-22T23:28:14.629+0100] {standard_task_runner.py:85} INFO - Job 80: Subtask parse_xml
[2023-11-22T23:28:14.676+0100] {task_command.py:416} INFO - Running <TaskInstance: xml_parsing.parse_xml scheduled__2023-11-22T22:23:00+00:00 [running]> on host yzpt-Lenovo-ideapad-320S-14IKB
[2023-11-22T23:28:14.725+0100] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='admin' AIRFLOW_CTX_DAG_ID='xml_parsing' AIRFLOW_CTX_TASK_ID='parse_xml' AIRFLOW_CTX_EXECUTION_DATE='2023-11-22T22:23:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-11-22T22:23:00+00:00'
[2023-11-22T23:28:16.622+0100] {logging_mixin.py:151} INFO - XML parsed successfully.
[2023-11-22T23:28:16.622+0100] {python.py:194} INFO - Done. Returned value was: <Element 'pdv_liste' at 0x7fb405b19990>
[2023-11-22T23:28:16.625+0100] {xcom.py:661} ERROR - Object of type Element is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your airflow config or make sure to decorate your object with attr.
[2023-11-22T23:28:16.625+0100] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/serialization/serde.py", line 178, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'xml.etree.ElementTree.Element'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/utils/session.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 2479, in xcom_push
    XCom.set(
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/utils/session.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/models/xcom.py", line 244, in set
    value = cls.serialize_value(
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/models/xcom.py", line 659, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/yzpt/projects/carburant_gcp/venv/lib/python3.10/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Element is not JSON serializable
[2023-11-22T23:28:16.629+0100] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=xml_parsing, task_id=parse_xml, execution_date=20231122T222300, start_date=20231122T222814, end_date=20231122T222816
[2023-11-22T23:28:16.642+0100] {standard_task_runner.py:104} ERROR - Failed to execute job 80 for task parse_xml (Object of type Element is not JSON serializable; 24037)
[2023-11-22T23:28:16.658+0100] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-11-22T23:28:16.674+0100] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
